src/install/repository.zig

```diff
@@ -151,8 +153,29 @@ pub const Repository = extern struct {
         env: *DotEnv.Loader,
         argv: []const string,
     ) !string {
+        // Note: currently if the user sets this to some value that causes
+        // a prompt for a password, the stdout of the prompt will be masked
+        // by further output of the rest of the install process.
+        // A value can still be entered, but we need to find a workaround
+        // so the user can see what is being prompted. By default the settings
+        // below will cause no prompt and throw instead.
+        const originalAskPass = env.map.get("GIT_ASKPASS");
+        const originalSSHCommand = env.map.get("GIT_SSH_COMMAND");
+
+        try env.map.putAllocKeyAndValue(allocator, "GIT_ASKPASS", "echo");
+        try env.map.putAllocKeyAndValue(allocator, "GIT_SSH_COMMAND", "ssh -oStrictHostKeyChecking=accept-new");
+
         var std_map = try env.map.stdEnvMap(allocator);
-        defer std_map.deinit();
+
+        defer {
+            _ = if (originalAskPass) |value| env.map.putAllocKeyAndValue(allocator, "GIT_ASKPASS", value) catch void
```

Comment:
Oh right, this is wrong. It won't update the key, forgot that this 'gets or updates'

Reply:
Updated. Am I right with my assumption here @dylan-conway that if the vars were replaced with putAllocKeyAndValue in the previous call - this won't revert it? I converted it to `.put` now.

Reply:
It will revert them, but `putAllocKeyAndValue` will also overwrite the existing vars on lines 165 and 166. We want to keep the existing ones, so instead `getOrPutWithoutValue` works better here. You could do something like:
```zig
const askpass_entry = env.map.getOrPutWithoutValue("GIT_ASKPASS") catch bun.outOfMemory();
if (!askpass_entry.found_existing) {
    askpass_entry.key_ptr.* = allocator.dupe(u8, "GIT_ASKPASS") catch bun.outOfMemory();
    askpass_entry.value_ptr.* = .{
        .value = allocator.dupe(u8, "echo") catch bun.outOfMemory(),
        .conditional = false,
    };
}
```
then in the `defer` block:
```zig
if (!askpass_entry.found_existing) {
    env.map.remove("GIT_ASKPASS");
}
```